toolchest/pyomoio.py

```diff
@@ -0,0 +1,217 @@
+"""pyomoio: read data from Pyomo models to pandas DataFrames
+
+Pyomo is a AMPL-like model description language for mathematical
+optimization problems. This module provides functions to read data from 
+Pyomo model instances and result objects. Use list_entities to get a list
+of all entities (sets, params, variables, objectives or constraints) inside a 
+pyomo instance, before get its contents by get_entity (or get_entities).
+    
+"""
+import pyomo.core as pyomo
+import pandas as pd
+
+def get_entity(instance, name):
```

Comment:
I'll suggest a single `get_entities` API rather than having two separate functions. You can inspect the `name` argument with

```
import collections

def iterable_not_string(test):
    return isinstance(test, collections.Iterable) \
        and not isinstance(test, basestring)

names = names if iterable_not_string(names) else [names]
```